include/unifex/type_traits.hpp

```diff
@@ -80,6 +80,41 @@ struct decayed_tuple {
 };
 
 template <typename T, typename... Ts>
-constexpr bool is_one_of_v = (std::is_same_v<T, Ts> || ...);
+inline constexpr bool is_one_of_v = (std::is_same_v<T, Ts> || ...);
+
+template <typename Fn, typename... As>
+using callable_result_t =
+    decltype(static_cast<Fn(*)() noexcept>(nullptr)()(
+             static_cast<As(*)() noexcept>(nullptr)()...));
+
+namespace detail {
+  template <typename Fn, typename... As>
+  inline constexpr bool _is_throw_callable_v =
+      noexcept(static_cast<Fn(*)() noexcept>(nullptr)()(
+               static_cast<As(*)() noexcept>(nullptr)()...));
+
+  template <
+      typename Fn,
+      typename... As,
+      typename = callable_result_t<Fn, As...>>
+  std::true_type _try_call(Fn(*)(As...))
+      noexcept(_is_throw_callable_v<Fn, As...>);
+  std::false_type _try_call(...);
+} // namespace detail
+
+template <typename Fn, typename... As>
+using is_callable =
+    decltype(detail::_try_call(static_cast<Fn(*)(As...)>(nullptr)));
+
+template <typename Fn, typename... As>
+inline constexpr bool is_callable_v = is_callable<Fn, As...>::value;
```

Comment:
In `is_tag_invocable_v` we used `yes_type` and `no_type` of different sizes and compared result sizes to determine the bool result rather than looking up true_type/false_type::value. Then we defined `is_tag_invocable` in terms of `is_tag_invocable_v`.

Should we do that here too? Or is it ok if we use bool_constant since it's probably going to be instantiated anyway?

Reply:
If you use any type trait, then `std::false_type` and `std::true_type` are instantiated.